packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -1,24 +1,16 @@
-import React, { useCallback, useEffect, useState } from 'react'
+import React, { useCallback, useState } from 'react'
 import { useDrag, useDrop } from 'react-dnd'
+import { RxDoubleArrowRight as ArrowRight, RxDoubleArrowDown as ArrowDown } from 'react-icons/rx'
 
+import { withContextMenu } from '../../hoc/withContextMenu'
 import { Input } from '../Input'
-import { Controls, Position } from '../Controls'
-import { RxDoubleArrowRight as ArrowRight, RxDoubleArrowDown as ArrowDown } from 'react-icons/rx'
+import { ContextMenu } from './ContextMenu'
 
 import './Tree.css'
 
-interface ContextMenu<T> {
-  value: T
-  position: Position
-}
-
-interface ContextMenuProps<T> {
-  contextMenu?: ContextMenu<T>
-  onContextMenuChange: (value?: ContextMenu<T>) => void
-}
-
 type Props<T> = {
   value: T
+  extraContextMenu: (value: T) => JSX.Element | null
```

Comment:
would you make this optional so it's not mandatory for all the trees to have an extra context menu getter?

Reply:
fixed 👍